src/compiler/diagnosticMessages.json

```diff
@@ -1823,6 +1823,22 @@
         "category": "Error",
         "code": 2671
     },
+    "Cannot assign a '{0}' constructor type to a '{1}' constructor type.": {
+        "category": "Error",
+        "code": 2672
+    },
+    "Constructor of type '{0}' is private and only accessible within class '{1}'.": {
+        "category": "Error",
+        "code": 2673
+    },
+    "Constructor of type '{0}' is protected and only accessible within class '{1}'.": {

```

Comment:
> `or a class derived from it.`

I have currently implemented it such that a class with a protected constructor cannot be instantiated within a derived class.
Doesn't this error message go against that? Or do I need to reimplement it such that it can be *accessible* within derived classes?